Romanchenko 181-1 Set with AVL tree

Romanchenko opened the pull request:
Я сделала сначала новый репозиторий для этой задачи, но там надо было опять принимать инвайт. Я его отправила, конечно, но осознала потом, что в хешмапе будет быстрее, видимо. 

https://neerc.ifmo.ru/wiki/index.php?title=%D0%90%D0%92%D0%9B-%D0%B4%D0%B5%D1%80%D0%B5%D0%B2%D0%BE - статья

https://official.contest.yandex.ru/contest/12783/run-report/20671354/

Romanchenko committed:
Set with AVL tree

Я сделала сначала новый репозиторий для этой задачи, но там надо было опять принимать инвайт. Я его отправила, конечно, но осознала потом, что в хешмапе будет быстрее, видимо. 

https://neerc.ifmo.ru/wiki/index.php?title=%D0%90%D0%92%D0%9B-%D0%B4%D0%B5%D1%80%D0%B5%D0%B2%D0%BE - статья

https://official.contest.yandex.ru/contest/12783/run-report/20671354/

glebshp reviewed (changes requested):
Привести в соответствие требованиям по комментариям (смотри чат).

http://www.cplusplus.com/reference/set/set/begin/
По стандарту begin() и end() должны работать за O(1)

Romanchenko committed:
Update set.h

Добавлены комментарии, исправлен begin() - теперь всегда поддерживается указатель на самый левый элемент.
Плюс картинки.
посылка - https://official.contest.yandex.ru/contest/12783/run-report/20718118/
